build/Dockerfile

```diff
@@ -4,13 +4,15 @@
 FROM alpine:3.15
 LABEL maintainer="offen <hioffen@posteo.de>"
 
+ARG TARGETARCH
+
 RUN addgroup -g 10001 -S offen \
 	&& adduser -u 10000 -S -G offen -h /home/offen offen
 RUN apk add -U --no-cache ca-certificates libcap tini bind-tools
 
-COPY ./bin/offen-linux-amd64 /opt/offen/offen-linux-amd64
-RUN setcap CAP_NET_BIND_SERVICE=+eip /opt/offen/offen-linux-amd64
-RUN ln -s /opt/offen/offen-linux-amd64 /sbin/offen
+COPY ./bin/offen-linux-$TARGETARCH /opt/offen/offen-linux-$TARGETARCH
```

Comment:
Did you check whether TARGETARCH matches what xgo builds for us when building `linux/arm/v7`? In case not we might need to manually rename the artifact so it matches what Docker expects here.

Reply:
I am unable to find what docker provides with the `$TARGETARCH` variable.
- [x] `--platform linux/amd64` -> `$TARGETARCH=amd64` (xgo produces `linux-amd64`) 
- [x] `--platform linux/arm64` -> `$TARGETARCH=arm64` (xgo produces `linux-arm64`)
- [ ] `--platform linux/arm-7` -> ??? (xgo produces `linux-arm-7`)

If they are consistent, it should be fine. but I'm **not** certain.

Reply:
I'll give it a test run later today.

Reply:
Hmm I tried building the binaries, which worked well, but I couldn't get the `arm/v7` Docker image to build correctly yet:

```
docker buildx build -f build/Dockerfile --platform linux/arm/v7 .
...
...
 => ERROR [4/8] COPY ./bin/offen-linux-arm /opt/offen/offen-linux-arm                                                                                                                                 0.0s
------
 > [4/8] COPY ./bin/offen-linux-arm /opt/offen/offen-linux-arm:
------
Dockerfile:13
--------------------
  11 |     RUN apk add -U --no-cache ca-certificates libcap tini bind-tools
  12 |     
  13 | >>> COPY ./bin/offen-linux-$TARGETARCH /opt/offen/offen-linux-$TARGETARCH
  14 |     RUN setcap CAP_NET_BIND_SERVICE=+eip /opt/offen/offen-linux-$TARGETARCH
  15 |     RUN ln -s /opt/offen/offen-linux-$TARGETARCH /sbin/offen
--------------------
error: failed to solve: failed to compute cache key: "/bin/offen-linux-arm": not found
```

The ugly part is what's coming next: apparently when passing `linux/arm/v7` as the `platform`, `TARGET_ARCH` will be `arm` (no v7 or similar). So I would guess we'll need to do the following still:
- Find out how that third (optional) chunk in the `platform` is called and if it is available in the context of the Dockerfile
- Rename the `offen-linux-arm-7` binary before building the Docker image so it can be found

---

Apart from that problem things seem to work well, I could build `amd64`, `arm64` and also `arm/v7` (after renaming manually) images just fine. This will be a great addition :tada:

Reply:
Seems it's called `TARGETVARIANT`: https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

Reply:
I've just updated the PR with `$TARGETVARIANT`. But I woudn't say it's pretty.. Let me know.
As it stands now. When the `$TARGETVARIANT` is set. It will be replaced with `-7` (hardcoded)